src/neon/deser.rs

```diff
@@ -0,0 +1,210 @@
+
+pub use crate::neon::intrinsics::*;
+
+unsafe fn find_bs_bits_and_quote_bits(src: &[u8]) -> ParseStringHelper {
+    // this can read up to 31 bytes beyond the buffer size, but we require
+    // SIMDJSON_PADDING of padding
+    let v0 : uint8x16_t = vld1q_u8(src.as_ptr());
+    let v1 : uint8x16_t = vld1q_u8(src.as_ptr().add(16));
+
+    let bs_mask : uint8x16_t = vmovq_n_u8('\\' as u8);
+    let qt_mask : uint8x16_t = vmovq_n_u8('"' as u8);
+
+    let bit_mask = uint8x16_t::new(0x01, 0x02, 0x4, 0x8, 0x10, 0x20, 0x40, 0x80,
+                                   0x01, 0x02, 0x4, 0x8, 0x10, 0x20, 0x40, 0x80);
+
+    let cmp_bs_0 : uint8x16_t = vceqq_u8(v0, bs_mask);
+    let cmp_bs_1 : uint8x16_t = vceqq_u8(v1, bs_mask);
+    let cmp_qt_0 : uint8x16_t = vceqq_u8(v0, qt_mask);
+    let cmp_qt_1 : uint8x16_t = vceqq_u8(v1, qt_mask);
+
+    let cmp_bs_0 = vandq_u8(cmp_bs_0, bit_mask);
+    let cmp_bs_1 = vandq_u8(cmp_bs_1, bit_mask);
+    let cmp_qt_0 = vandq_u8(cmp_qt_0, bit_mask);
+    let cmp_qt_1 = vandq_u8(cmp_qt_1, bit_mask);
+
+    let sum0 : uint8x16_t = vpaddq_u8(cmp_bs_0, cmp_bs_1);
+    let sum1 : uint8x16_t = vpaddq_u8(cmp_qt_0, cmp_qt_1);
+    let sum0 = vpaddq_u8(sum0, sum1);
+    let sum0 = vpaddq_u8(sum0, sum0);
+
+    ParseStringHelper {
+        bs_bits: vgetq_lane_u32(vreinterpretq_u32_u8(sum0), 0), // bs_bits
+        quote_bits: vgetq_lane_u32(vreinterpretq_u32_u8(sum0), 1)  // quote_bits
+    }
+}
+
+impl<'de> Deserializer<'de> {
+    #[cfg_attr(not(feature = "no-inline"), inline(always))]
+    pub fn parse_str_(&mut self) -> Result<&'de str> {
+        // Add 1 to skip the initial "
+        let idx = self.iidx + 1;
+        let mut padding = [0u8; 32];
+        //let mut read: usize = 0;
+
+        // we include the terminal '"' so we know where to end
+        // This is safe since we check sub's lenght in the range access above and only
+        // create sub sliced form sub to `sub.len()`.
+
+        let src: &[u8] = unsafe { &self.input.get_unchecked(idx..) };
+        let mut src_i: usize = 0;
+        let mut len = src_i;
+        loop {
+            // store to dest unconditionally - we can overwrite the bits we don't like
+            // later
+
+            let srcx = if src.len() >= src_i + 32 {
+                &src[src_i..]
+            } else {
+                unsafe {
+                    padding
+                        .get_unchecked_mut(..src.len() - src_i)
+                        .clone_from_slice(src.get_unchecked(src_i..));
+                    &padding
+                }
+            };
+
+            let ParseStringHelper { bs_bits, quote_bits } = unsafe { find_bs_bits_and_quote_bits(&srcx) };
+
+            if (bs_bits.wrapping_sub(1) & quote_bits) != 0 {
+                // we encountered quotes first. Move dst to point to quotes and exit
+                // find out where the quote is...
+                let quote_dist: u32 = trailingzeroes(u64::from(quote_bits)) as u32;
```

Comment:
While looking at this I found that we can just use `.trailing_zeros()` here. It doesn't make sense to change it everywhere in this PR as it would conflate concerns, but we might as well start with the arm code to clean up some of those things :)

Reply:
In arm this would also remove the need for conversion to u64 I believe.

Reply:
@Licenser maybe I'll sneak a PR in ahead of this for the `.trailing_zeros()` usage on the x86 side?

Reply:
sounds good! 👍